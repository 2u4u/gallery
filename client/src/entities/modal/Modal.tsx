import { useEffect } from 'react';
import './index.scss';

type Props = {
    handleCloseModal: () => void;
    children: React.ReactNode;
    title: string;
}

function Modal({ handleCloseModal, children, title }: Props) {
    useEffect(() => {
        document.body.classList.add('overflow-hidden');
        return () => {
            document.body.classList.remove('overflow-hidden');
        }
    })
    
    return (
        <div className='modal'> 
            <div className='modal__backdrop'></div>
            <div className='modal__content'>
                <h2 className='modal__header'>{title}</h2>
                <button className='modal__close' onClick={handleCloseModal}>
                    <img src="./img/close.svg" alt="close modal" />
                </button>
                {children}
            </div>
        </div>
    );
}

export default Modal;
