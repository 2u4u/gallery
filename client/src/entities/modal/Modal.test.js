import renderer from 'react-test-renderer';
import Modal from './Modal';

describe('Modal component ', () => {
    it('matches snapshot', () => {
        const component = renderer.create(
            <Modal title='Hello world' handleCloseModal={() => false}>
                <div>some content</div>
            </Modal>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
