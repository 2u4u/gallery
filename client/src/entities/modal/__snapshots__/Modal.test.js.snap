// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component  matches snapshot 1`] = `
<div
  className="modal"
>
  <div
    className="modal__backdrop"
  />
  <div
    className="modal__content"
  >
    <h2
      className="modal__header"
    >
      Hello world
    </h2>
    <button
      className="modal__close"
      onClick={[Function]}
    >
      <img
        alt="close modal"
        src="./img/close.svg"
      />
    </button>
    <div>
      some content
    </div>
  </div>
</div>
`;
